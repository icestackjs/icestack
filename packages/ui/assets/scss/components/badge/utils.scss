@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("badge.selector");
$DefaultMap: inject("badge.defaults.utils");

#{$S} {
  @each $typeName, $map in map.get($DefaultMap, "sizes") {
    &#{getSelector($typeName)} {
      @include resolve(map.get($map, "default"));
    }
  }
}
