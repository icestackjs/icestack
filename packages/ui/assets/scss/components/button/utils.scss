@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("button.selector");
$DefaultMap: inject("button.defaults.utils");

#{$S} {
  @each $size, $map in map.get($DefaultMap, "sizes") {
    &#{getSelector($size)} {
      @include resolve(map.get($map, "default"));
    }
  }

  @each $shape, $map in map.get($DefaultMap, "shapes") {
    &#{getSelector($shape)} {
      @include resolve(map.get($map, "default"));

      @each $size, $m in map.get($map, "sizes") {
        &:where(#{getSelector($size,#{$S}-)}) {
          @include resolve(map.get($m, "default"));
        }
      }
    }
  }
}
