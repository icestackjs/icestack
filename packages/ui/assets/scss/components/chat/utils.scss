@use "../../utils/index.scss" as *;

@include spread(inject("chat.defaults.utils"));
