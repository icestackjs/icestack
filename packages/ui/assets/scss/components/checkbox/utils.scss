@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("checkbox.selector");
$DefaultMap: inject("checkbox.defaults.utils");

@mixin utils {
  @each $s, $v in map.get($DefaultMap, "sizes") {
    &#{getSelector($s)} {
      @include resolve(map.get($v, "default"));
    }
  }
}

#{$S} {
  @include utils;
}
