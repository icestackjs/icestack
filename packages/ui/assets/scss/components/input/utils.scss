@use "../../utils/index.scss" as *;

@include spread(inject("input.defaults.utils"));
