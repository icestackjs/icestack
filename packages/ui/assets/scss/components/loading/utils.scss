@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("loading.selector");
$DefaultMap: inject("loading.defaults.utils");

#{$S} {
  @each $s, $v in map.get($DefaultMap, "sizes") {
    &#{getSelector($s)} {
      @include resolve($v);
    }
  }

  @each $s, $v in map.get($DefaultMap, "shapes") {
    &#{getSelector($s)} {
      @include resolve($v);
    }
  }
}
