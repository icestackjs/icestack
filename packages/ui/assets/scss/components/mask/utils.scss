@use "../../utils/index.scss" as *;
@use "sass:map";
$S: inject("mask.selector");
$DefaultMap: inject("mask.defaults.utils");

#{$S} {
  @each $s, $v in map.get($DefaultMap, "shapes") {
    &#{getSelector($s)} {
      @include resolve($v);
    }
  }
}
