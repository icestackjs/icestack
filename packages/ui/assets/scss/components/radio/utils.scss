@use "../../utils/index.scss" as *;
@use "sass:map";

$D: inject("radio.defaults.utils");
$S: inject("radio.selector");

#{$S} {
  @each $s, $v in map.get($D, "sizes") {
    &#{getSelector($s)} {
      @include resolve(map.get($v, "default"));
    }
  }
}
