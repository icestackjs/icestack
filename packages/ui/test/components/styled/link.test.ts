import { resolve } from './utils'
import { compileScss } from '@/sass'
describe('link', () => {
  it('snap', async () => {
    const css = await compileScss(resolve('link'))
    expect(css).toMatchSnapshot()
  })
})
