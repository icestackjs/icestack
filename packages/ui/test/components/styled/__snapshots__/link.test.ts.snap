// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`link > snap 1`] = `
".link-primary {
  @apply text-primary [@media(hover:hover)]:hover:text-primary-focus;
}
.link-info {
  @apply text-info [@media(hover:hover)]:hover:text-info-focus;
}
.link-success {
  @apply text-success [@media(hover:hover)]:hover:text-success-focus;
}
.link-warning {
  @apply text-warning [@media(hover:hover)]:hover:text-warning-focus;
}
.link-error {
  @apply text-error [@media(hover:hover)]:hover:text-error-focus;
}
.link-neutral {
  @apply text-neutral [@media(hover:hover)]:hover:text-neutral-focus;
}
.link:focus {
  @apply outline-none;
}
.link:focus-visible {
  outline: 2px solid currentColor;
  outline-offset: 2px;
}"
`;
